### DISCUSSION TITLE
How is the audio visualization supposed to be used with an HDMI grabber?

### DISCUSSION THREAD

**Discussion by protyposis** [awawa-dev/HyperHDR#754] [Q&A] [answered]
HyperHDR allows configuring audio background effects for when it is in idle mode. As I understand, idle mode means that there is no video signal. However, with an HDMI grabber, I always have either audio and video at the same time, or no signal at all. So, whenever I have an audio signal, HyperHDR is not in idle mode (because there is also a video signal).

My current workaround is (1) reconfiguring the background effect between a static color (which I display when there is no HDMI signal) and an audio effect, and (2) selecting the appropriate source to toggle between audio and video visualization.

This feels a little complicated and makes me wonder if I am missing something. Is there a better solution to this?

**Comment by awawa-dev** [comment]
> HyperHDR allows configuring audio background effects for when it is in idle mode. As I understand, idle mode means that there is no video signal. 

You can use signal detection: manual or automatic in the grabber option. HyperHDR will switch to the background effect (aka `idle mode`) even if there is theoretically a video source, but it only consists of a static, identical incoming frames (usually 'no-signal' board grabber: black screen or rainbow test board as in ms2109).

**Reply by protyposis** [reply]
This doesn't work because the apps still display a kind of dynamic picture (e.g. moving background pictures or colors, playback time indicators, etc.). If it worked, it would lead to false positives: e.g., when the Android TV home screen is displayed, which is actually static, it would unexpectedly switch to audio visualization. Also, this doesn't seem to help me switch between an audio visualization mode and "true-idle" mode (with static background color)?

**Reply by awawa-dev** [reply, marked as answer]
The background effect is not the only place where you can set audio effects. They can be started directly in the 'remote' tab, even if a video stream is present and it behaves exactly like any other graphic effect in terms of highest priority (so no need to manually switching source) and enabling/disabling with the main interface or API (JSON or MQTT). That's all.

**Reply by protyposis** [reply]
This is exactly what I was missing, thank you! Works great 🥳
